DSPPlugCombinedVolcanoPlot/LabeledCombinedVolcanoPlot.R

```diff
@@ -0,0 +1,648 @@
+                                paste(label_thresh, positive_label))
+    }
+    
+    color_label <- "Significance:"
+    
+    # color by fc_thresh if fc_thresh is not NULL
+    if(!is.null(fc_thresh)){
+      gene_coloring$Target_coloring[abs(gene_coloring$Log2) < fc_thresh] <- paste("FC <", fc_thresh)
+      color_options <- c(color_options, fc_color)
+      names(color_options)[length(color_options)] <- paste("FC <", fc_thresh)
+    }
+  }
+  if(color_by_dataset == FALSE){ #for other points without colors specified  
+  color_options <- c(color_options, default_color)
+  names(color_options)[length(color_options)] <- "Not Specified"
+  }
+  # add coloring to ggplot
+  ggFigure <- ggFigure + geom_point(data=gene_coloring, aes(x=Log2, y=Pvalue, color=Target_coloring))+
+    labs(color=color_label)+
+    scale_color_manual(values=color_options)
+  
+  # add threshold line values to y axis
+  yaxis <- data.frame(brk=as.numeric(pretty_breaks(n=4)(0:max(de$X.log10.pvalue))))
+  yaxis$brk <- 10^-(yaxis$brk)
+  
+  # keep scientific notation if small enough pvalues when changing to character
+  yaxis$label <- format(yaxis$brk, trim=TRUE, digits=4,
+                        scientific=ifelse(maxPval < 0.0001, TRUE, FALSE), 
+                        drop0trailing=TRUE)
+  
+  # add threshold lines if thresholds are not NULL
+  if(!is.null(fc_thresh)){
+    ggFigure <- ggFigure + geom_vline(xintercept=fc_thresh, linetype="dotted")+
+      geom_vline(xintercept=-fc_thresh, linetype="dotted")+
+      annotate("text", x=fc_thresh+0.35, y=1,family=font_family, size=font_size/3,
+               label=paste0("FC=", round(2^fc_thresh, digits=2)))
+  }
+  if(!is.null(pval_thresh)){
+    ggFigure <- ggFigure + geom_hline(yintercept=pval_thresh, linetype="dotted")
+    yaxis <- rbind(yaxis, c(pval_thresh, paste0('pval=',pval_thresh)))
+  }
+  if(!is.null(fdr_thresh)){
+    ggFigure <- ggFigure + geom_hline(yintercept=fdr_pval, linetype="dotted")
+    yaxis <- rbind(yaxis, c(fdr_pval, paste0('FDR=',fdr_thresh)))
+  }
+  
+  # order yaxis in increasing value 
+  yaxis$brk <- as.numeric(yaxis$brk)
+  yaxis <- yaxis[order(yaxis$brk, decreasing=F),]
+  
+  # subset de to only contain genes to label on plot, either by user specified genes or top n_genes by pval
+  if(!is.null(gene_list)){
+    gene_labels <- subset(de, subset=Target.Name %in% gene_list)
+  }else{
+    # remove gene labels for genes below lowest pvalue cutoff
+    gene_labels <- de[which(de$Pvalue < min(fdr_pval, pval_thresh, na.rm = T)),]
+    
+    # only label genes above fc_thresh if set by user else only look at pvalue
+    if(!is.null(fc_thresh) & label_fc == FALSE){
+      gene_labels <- gene_labels[which(abs(gene_labels$Log2) > fc_thresh),]
+    }
+    
+    # only keep top # of genes by pvalue
+    gene_labels <- gene_labels[head(order(gene_labels$Pvalue, decreasing=FALSE), n=n_genes),]
+  }  
+  interactive=ifelse(output_format=="html", TRUE, FALSE)
```

Comment:
I'm confused by the html output, is it allowed or not? It seems to be allowed here but not everywhere else. 

specifically 
- commenting out the saving of HTML on L209-210
- not be listed as a valid output type in docs and L15

Reply:
It requires html widgets to download, so it will not work in dspda. I commented out so it is easy to turn on when the packages are installed in DSPDA.